### PR TITLE
Fix tests and introduce elasticsearch polling for apiman 2.0 release

### PR DESCRIPTION
**Pull request by bekihm**
This is a test branch for the 2.0 release. We will test if there is a flushing problem while updating the data entities for the EsStorage.

### CODE REVIEW

**Review by volkflo — APPROVED** [review]

**Comment by bekihm** [comment]
If this PR solves the problem with the tests we will fix it for all other cases (index and delete entities). So we will close this PR and do a PR doing all things together.

**Comment by bekihm** [comment]
So now I added also a new polling improvement for the elasticsearch client. This improves also the test framework code.
I the tests are green we can merge it after the review and do the apiman 2.0 release :-)

File: .github/workflows/verify.yaml [hunk 1]
```diff
@@ -36,7 +36,10 @@ jobs:
       - name: Run Tests (Vert.x 3 Mem)
         run: mvn test install -Dinvoker.skip -Dmaven.javadoc.skip=true -B -Dapiman.gateway-test.config=vertx3-mem
       - name: Run Tests (Vert.x 3 ES)
-        run: mvn test install -Dinvoker.skip -Dmaven.javadoc.skip=true -B -Dapiman.gateway-test.config=vertx3-es
+        run: |
+          docker run -d -p 19250:9200 -p 9300:9300 -e "discovery.type=single-node" --name=elasticsearch-7 docker.elastic.co/elasticsearch/elasticsearch-oss:7.6.2
```
Comment (volkflo) [inline .github/workflows/verify.yaml:40]: Would be cool to read this version in future out of the pom.xml

File: manager/api/es/src/main/java/io/apiman/manager/api/es/EsStorage.java [hunk 2]
```diff
@@ -2308,6 +2210,14 @@ public ApiVersionBean unmarshal(Map<String, Object> source) {
         }, qb);
     }
 
+    /**
+     * Refreshes an index
+     * @param indexPostfix the index postfix name
+     */
+    public void refresh(String indexPostfix) throws IOException {
```
Comment (volkflo) [inline manager/api/es/src/main/java/io/apiman/manager/api/es/EsStorage.java:2217]: Is this method still in use @bekihm ?

**Review by volkflo — COMMENTED** [review]
LGTM

File: manager/api/es/src/main/java/io/apiman/manager/api/es/EsStorage.java (hunk 2 above)
Comment (bekihm) [reply manager/api/es/src/main/java/io/apiman/manager/api/es/EsStorage.java:2217]: Yes it is used for the tests from classes like TestEsStorageWrapper and TestEsStorageQueryWrapper

File: .github/workflows/verify.yaml (hunk 1 above)
Comment (bekihm) [reply .github/workflows/verify.yaml:40]: i agree with you

**Comment by bekihm** [comment]
@EricWittmann I think we are ready :-)

**Comment by EricWittmann** [comment]
image

Oh yeah!

File: .github/workflows/verify.yaml (hunk 1 above)
Comment (EricWittmann) [inline .github/workflows/verify.yaml:40]: Why is this not needed for the `servlet-es` test variant?

**Review by EricWittmann — APPROVED** [review]

File: .github/workflows/verify.yaml (hunk 1 above)
Comment (bekihm) [reply .github/workflows/verify.yaml:40]: I will take a further look tomorrow, the tests for `vert.x` and `servlet-es` work not in a consistent way. Historically I think the embedded elasticsearch library was only used in the `servlet-es` variant. But I will check tomorrow. Maybe we can improve the test framework further, this was not part of our elasticsearch upgrade so far.

Comment (EricWittmann) [reply .github/workflows/verify.yaml:40]: It wasn't really a request for a change, more a matter of curiosity.  :)

**Pull request merged**